### PR TITLE
fix: Prevent type parsing to interfere with package name in typescrip…

### PR DESCRIPTION
**Pull request by ValentinTrinque**
Before that fix:

The method...

```go
func (h *Handler) RespondToInteraction(interaction interactor.Interaction) {}
```

... would generate...

```ts
export function RespondToInteraction(arg1:number):Promise<Error>;
```

... because the `interaction` package starts with `int` and anything starting with `int` is interpreted as `number` as defined here:

https://github.com/wailsapp/wails/blob/eee67979078f65887e90c5d60084af1c22837960/v2/internal/binding/generate.go#L130-L134

### CODE REVIEW

**Review by leaanthony — COMMENTED** [review]
Nice find and thanks for the PR! We are encouraging everyone to include tests with PRs to alert us to regressions at a later date. Do you think you could add a test to this PR? We have tests defined in `v2/internal/binding/generate_test.go`.

Thanks 🙏

**Comment by ValentinTrinque** [comment]
Yes, of course, I quickly tried, but testing this part is a not super straight forward. I'll see what I can do

**Comment by ValentinTrinque** [comment]
In the end, that was pretty easy.

**Review by leaanthony — COMMENTED** [review]
Thanks for this. Added a comment. Thoughts?

File: v2/internal/binding/binding_test/binding_conflicting_package_name_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+func TestConflictingPackageName(t *testing.T) {
+	// given
+	generationDir := t.TempDir()
+
+	// setup
+	testLogger := &logger.Logger{}
+	b := binding.NewBindings(testLogger, []interface{}{&HandlerTest{}}, []interface{}{}, false)
+
+	// then
+	err := b.GenerateGoBindings(generationDir)
```
[the remark this answers is not in the document]
Comment (ValentinTrinque) [reply v2/internal/binding/binding_test/binding_conflicting_package_name_test.go:48]: Yes, that would have been easier.

I think we could significantly improve the design (and its _testability_) by splitting the bindings generation into multiple component (with Single Responsibility Principle, in mind).

I already had a look to hit, but wanted to keep this fix refactoring-free.

Here is just my brain dump on the matter.

Basically, the idea would be:

1. A component that parse the go code and build an AST. That already exist AFAIK.
2. A component that translate that Go AST into a JS AST
3. Another component that translate that Go AST into TS AST
4. A component responsible to write the JS AST as JS code
5. Another component responsible to write the TS AST as TS code


The binding code would essentially become, as pseudo code:

```go
func main() {
  writer := fileWriter{}

  goAST := GetGoASTFromBoundStructs({}interface{&Handler{}})

  go func() {
    ast := toJavascriptAST(goAST)
    obfuscatedAST := obsfuscateJavascriptAST(ast)
    code := toJavascriptCode(jsAST)
    writer.Write(code, "Handler.js")
  }()

  go func() {
    ast := toTypescriptAST(goAST)
    obfuscatedAST := obsfuscateTypescriptAST(ast)
    code := toTypescriptCode(ast)
    writer.Write(jsCode, "Handler.d.ts")
  }()
}
```

That example doesn't take into account the model generation, UPX and all the nitty gritty details. It's just to convey the idea.

Such design definitely will add more loop interaction but I believe the testability, extensibility (adding specific processing, features, etc) and ability to replace each component will increase dramatically.

Also, this is just a draft thought as I haven't looked in the details. I think I saw that part of the existing code is used for Wails runtime to call the methods and all, so it really needs more thoughts and design.

I am of course open to critics :)

**Review by leaanthony — APPROVED** [review]
Let's look at refactoring separately 👍

**Comment by ValentinTrinque** [comment]
Thanks @leaanthony !

I opened an issue about the refactoring proposal #1953.

**Pull request merged**